simcon_project/static/js/recorder.js

```diff
@@ -0,0 +1,173 @@
+// pauseButton.addEventListener("click", pauseRecording);
+
+// Misc elements
+
+function startRecording() {
+	console.log("recordButton clicked");
+
+	/*
+		Simple constraints object, for more advanced audio features see
+		https://addpipe.com/blog/audio-constraints-getusermedia/
+	*/
+
+    var constraints = { audio: true, video:false }
+
+ 	/*
+    	Disable the record button until we get a success or fail from getUserMedia()
+	*/
+
+	recordButton.disabled = true;
+	stopButton.disabled = false;
+	// pauseButton.disabled = false
+
+	/*
+    	We're using the standard promise based getUserMedia()
+    	https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
+	*/
+
+	navigator.mediaDevices.getUserMedia(constraints).then(function(stream) {
+		console.log("getUserMedia() success, stream created, initializing Recorder.js ...");
+
+		/*
+			create an audio context after getUserMedia is called
+			sampleRate might change after getUserMedia is called, like it does on macOS when recording through AirPods
+			the sampleRate defaults to the one set in your OS for your playback device
+		*/
+		audioContext = new AudioContext();
+
+		//update the format
+		// document.getElementById("formats").innerHTML="Format: 1 channel pcm @ "+audioContext.sampleRate/1000+"kHz"
+
+		/*  assign to gumStream for later use  */
+		gumStream = stream;
+
+		/* use the stream */
+		input = audioContext.createMediaStreamSource(stream);
+
+		/*
+			Create the Recorder object and configure to record mono sound (1 channel)
+			Recording 2 channels  will double the file size
+		*/
+		rec = new Recorder(input,{numChannels:1})
+
+		//start the recording process
+		rec.record()
+
+		console.log("Recording started");
+
+	}).catch(function(err) {
+	  	//enable the record button if getUserMedia() fails
+    	recordButton.disabled = false;
+    	stopButton.disabled = true;
+    	// pauseButton.disabled = true
+	});
+}
+
+// function pauseRecording(){
+// 	console.log("pauseButton clicked rec.recording=",rec.recording );
+// 	if (rec.recording){
+// 		//pause
+// 		rec.stop();
+// 		pauseButton.innerHTML="Resume";
+// 	}else{
+// 		//resume
+// 		rec.record()
+// 		pauseButton.innerHTML="Pause";
+//
+// 	}
+// }
+
+function stopRecording() {
+	console.log("stopButton clicked");
+
+	//disable the stop button, enable the record too allow for new recordings
+	stopButton.disabled = true;
+	recordButton.disabled = true;
+	// pauseButton.disabled = true;
+
+	document.getElementById("choice-form").style.display = "block";
+	document.getElementById("embedded-video").style.display = "none";
+
+	//reset button just in case the recording is stopped while paused
+	// pauseButton.innerHTML="Pause";
+
+	//tell the recorder to stop the recording
+	rec.stop();
+
+	//stop microphone access
+	gumStream.getAudioTracks()[0].stop();
+
+	//create the wav blob and pass it on to createDownloadLink
+	rec.exportWAV(createDownloadLink);
+}
+
+function createDownloadLink(blob) {
```

Comment:
are we storing it in tatiana's computer or a db

Reply:
The way the 2014 project did it was by storing audio files in the project media folder, but they just gitignore all .wav files